src/nnet3/nnet-chain-example.cc

```diff
@@ -428,11 +428,17 @@ bool NnetChainExampleStructureCompare::operator () (
   for (size_t i = 0; i < size; i++)
     if (!io_compare(a.inputs[i], b.inputs[i]))
       return false;
+
```

Comment:
It would be more efficient if you somehow included the weight in the return of NnetChainExampleStructureHasher.  Otherwise it will have to do a lot of comparisons if people use many different weights.

Reply:
True, I'll try to figure something out.